Update contributor's checklists in PR template

ZanSara opened the pull request:
**Problem**:
- In external PRs, contributor's seems to often be confused about which checkboxes to tick. In some cases they ignore the list, in others they go ahead and check boxes that we assume were dedicated to reviewers. This makes the checklist itself less effective.

**Solution**:
- I believe splitting the checklist into Contributor's and Reviewer's can bring more clarity and ease the reviewer's job.
- Some checkboxes seemed outdated and therefore removed (the first draft vs final code: now PRs can be marked as Draft, by the reviewers as well if necessary)
- Some checkboxes were added for the reviewers (the labels review)

**Contributor's Checklist**:
- [ ] Relevant issue(s):
- [ ] [Enable actions on your fork](https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md#forks)
- [ ] Add tests
- [ ] Update docstrings

**Reviewer's Checklist**:
- [ ] Review labels
- [ ] Documentation PR (if relevant): (copy here link to PR from the documentation repo)

masci reviewed (changes requested):
Thanks for taking care of this!

In the spirit of this PR, I'd go all-in with making the PR template contributor-centric by removing any instruction for the reviewer (in any case, the item about documentation PRs will become obsolete very soon, which leaves the checklist with just one item).

About the contributor part, I'd rephrase it in a way that the recipient of the checklist is obvious, something like:

## Pre-flight checklist
- [ ]  I have read the [contributors guidelines](https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md)
- [ ] I have enabled actions on my fork
- [ ] If this is a code change, I have added tests and docstrings
- [ ] If this is a substantial change, the PR description contains a link to the relevant issue(s)

WDYT?

ZanSara commented:
I have slightly modified your template to keep tests and docstrings separate, and I moved the issues in the top to make it more prominent. It now looks like:

-------------------------------

**Related Issue(s)**:  ...

**Proposed changes**:
- ...

## Pre-flight checklist
- [ ]  I have read the [contributors guidelines](https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md)
- [ ] I have [enabled actions on my fork](https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md#forks)
- [ ] If this is a code change, I added tests or updated existing ones 
- [ ] If this is a code change, I updated the docstrings
